lib/membrane_tcp/sink.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Membrane.TCP.Sink do
+  @moduledoc """
+  Element that sends buffers received on the input pad over a TCP socket.
+  """
+  use Membrane.Sink
+
+  alias Membrane.Buffer
+  alias Membrane.TCP.{CommonSocketBehaviour, Socket}
+
+  def_options connection_side: [
```

Comment:
how about allowing to provide `server_address` and `server_pot_no` only in case of `connection_side: :client`? the `connection_side` argument could be somehow parametrized and equal to either `:server` or `{:client, server_address: <address>, server_port_no: <port>}`.
In such a case the parameters would be more tightly coupled and setting their default values would be easier.

Reply:
actually there is even more complexity, since when providing an already connected sockets the local_address and local_port_no options are ignored. An option accurately representing the state of affairs would have a spec like this:
```
spec:
  %Socket{state: :connected}
  | %{
    connection_side => :client,
    server_address => :inet.ip_address(),
    server_port_no => :inet.port_number(),
    local_address => :inet.socket_address(),
    local_port_no => :inet.port_number()
  } 
  | %{
    connection_side => :server,
    local_address => :inet.socket_address(),
    local_port_no => :inet.port_number()
  }
```
But I believe this is too much, defaults are really hard to set etc. I think we can settle for your suggestion of having connection_side have information about server address and port and in case of providing an already connected socket just ignore the local address and port. However in this case, when providing the socket and requiring the user to pass `{:client, address, port}` just for the port and address to be ignored is kinda iffy. I'd consider having an option of passing just `:client` to `connection_side`, that would require an already connected socket to be passed

Reply:
as discussed, we can use `Bunch.Config.parse` for that purpose ;)